.github/workflows/mkdocs.yml

```diff
@@ -0,0 +1,36 @@
+# This GitHub Actions workflow contains two jobs:
+# -- First, Lint uses the latest Ubuntu image and Python 3.9 to check
+# the code and writing for defects, using the Poetry lint task
+#
+# -- Second, Test uses a strategy matrix to test the code with different
+# Python versions and determine code coverage, using the Poetry test task
+
+name: Lint and Test
+
+on: [push, pull_request]
+
+jobs:
+  mkdocs:
+    name: MkDocs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Python 3.9
+        uses: actions/setup-python@v2
+        id: setup-python
+        with:
+          python-version: 3.9
+      - name: Install Poetry
+        uses: abatilo/actions-poetry@v2.1.4
+        with:
+          poetry-version: 1.1.13
+      - name: Setup Poetry
+        run: |
+          poetry config virtualenvs.create true
+          poetry config virtualenvs.in-project true
+          poetry env info
+      - name: Install dependencies
+        run: poetry install --no-interaction --no-ansi
+      - name: Update MkDocs
+        run: poetry run task mkdocs
```

Comment:
What is the planned functionality for this task? My followup on this is this: do you think that this task should run on every push and pull request for every branch or only when pushing/merging to main?

Reply:
Updated this to only run on merges to main! 👍🏻